### PR TITLE
Add mixed array and object dot notation support Closes: #47

### PR DESCRIPTION
**Pull request by AlexMeah**

### CODE REVIEW

File: lib/parse.js [hunk 1]
```diff
@@ -83,6 +83,10 @@ internals.parseKeys = function (key, val, options) {
         return;
     }
 
+    // Transform dot notation to bracket notation
+
+    key = key.replace(/\.(\w+)/g, '[$1]');
```
Comment (nlf) [inline lib/parse.js]: This feels too strict, `\w` is alphanumeric plus underscores, so it rules out plenty of valid characters. Maybe better to make it `/\.([^\.\[]+)/`.

I'm also curious how big of a hit this has on performance..

**Comment by nlf** [comment]
I'm trying to think of scenarios this might not work, but this is definitely the simplest solution for 90+% of use cases.

I'm still undecided how flexible I want this module to be, since there's been a *ton* of bikeshedding and debate over what it should do and how. Part of me thinks it should support as many use cases as possible, but the other part of me realizes it borders on crazy to do that and I should just keep what it does clearly defined and move on.

I'm still chewing on this, and I'll get back to it, but thanks for the effort so far!

File: lib/parse.js (hunk 1 above)
Comment (AlexMeah) [inline lib/parse.js]: I have updated the regex.

What is the best way to measure perf? A few console logs show the speed to be the same as before.

**Comment by AlexMeah** [comment]
We bumped into this issue when dealing with a legacy .net application which has to remain unchanged. 

We need to being accepting form submissions using express-body-parser and would like to avoid using a custom implementation on specific routes.

**Comment by Nijikokun** [comment]
:+1:

**Comment by AlexMeah** [comment]
Any movement on this?

**Comment by hueniverse** [comment]
@nlf this is a 4 months old pull request. Please merge or close.

**Pull request merged**